src/frontend/src/expr/table_function.rs

```diff
@@ -226,6 +229,132 @@ impl TableFunction {
         })
     }
 
+    pub fn new_postgres_query(args: Vec<ExprImpl>) -> RwResult<Self> {
+        let evaled_args = {
+            let mut evaled_args: Vec<String> = Vec::with_capacity(6);
+            if args.len() != 6 {
+                return Err(BindError("table_scan function only accepts 6 arguments: postgres_query(hostname varchar, port varchar, username varchar, password varchar, database_name varchar, postgres_query varchar)".to_string()).into());
+            }
+            for arg in &args {
+                if arg.return_type() != DataType::Varchar {
+                    return Err(BindError(
+                        "table_scan function only accepts string arguments".to_string(),
+                    )
+                    .into());
+                }
+                match arg.try_fold_const() {
+                    Some(Ok(value)) => {
+                        if value.is_none() {
+                            return Err(BindError(
+                                "table_scan function does not accept null arguments".to_string(),
+                            )
+                            .into());
+                        }
+                        match value {
+                            Some(ScalarImpl::Utf8(s)) => {
+                                evaled_args.push(s.to_string());
+                            }
+                            _ => {
+                                return Err(BindError(
+                                    "table_scan function only accepts string arguments".to_string(),
+                                )
+                                .into())
+                            }
+                        }
```

Comment:
nit: with the data type check above we can just let `into_utf8`/`as_utf8` panic:
```
let Some(scalar) = value else {
  return Err(BindError(
    "table_scan function does not accept null arguments".to_string(),
  )
  .into());
};
evaled_args.push(value.into_utf8().into());
```